litellm/proxy/guardrails/guardrail_hooks/acuvity.py

```diff
@@ -0,0 +1,173 @@
+import os
+from typing import Literal, Optional, Union
+
+from acuvity import Acuvity, GuardConfig, GuardName, ResponseMatch, Security
+from fastapi import HTTPException
+
+from litellm._logging import verbose_proxy_logger
+from litellm.caching.caching import DualCache
+from litellm.integrations.custom_guardrail import (
+    CustomGuardrail,
+    log_guardrail_information,
+)
+from litellm.litellm_core_utils.logging_utils import (
+    convert_litellm_response_object_to_str,
+)
+from litellm.proxy._types import UserAPIKeyAuth
+
+GUARDRAIL_NAME = "acuvity"
+
+class AcuvityGuardrail(CustomGuardrail):
+    def __init__(
+        self, api_key: Optional[str] = None, api_base: Optional[str] = None,
+        **kwargs,
+    ):
+        try:
+            if kwargs.get("event_hook") == "pre_call":
+                self.pre_guard_config_dict = kwargs.pop("vendor_params", None)  # None as default if not found
+                self.pre_guard_config = GuardConfig(self.pre_guard_config_dict)
```

Comment:
we should add a check here: only guards that can be redacted should be able to be specified here. And we also need to check that they are set with redaction enabled. Otherwise it can be considered a wrong configuration and we can fail the init here.